client-go/controller/client/http.go

```diff
@@ -81,20 +83,57 @@ func (c Client) Request(method string, path string, body []byte) (*http.Response
 }
 
 // BasicRequest makes a simple http request on the controller.
-func (c Client) BasicRequest(method string, path string, body []byte) (string, int, error) {
+func (c Client) BasicRequest(method string, path string, body []byte) (string, error) {
 	res, err := c.Request(method, path, body)
 
 	if err != nil {
-		return "", -1, err
+		return "", err
 	}
 	defer res.Body.Close()
 
 	resBody, err := ioutil.ReadAll(res.Body)
 
 	if err != nil {
-		return "", -1, err
+		return "", err
 	}
-	return string(resBody), res.StatusCode, nil
+	return string(resBody), checkForErrors(res, string(resBody))
+}
+
+func checkForErrors(res *http.Response, body string) error {
+	switch res.StatusCode {
+	case http.StatusOK, http.StatusCreated, http.StatusNoContent:
```

Comment:
I think so. Anything the API throws at us in the future should work. Checking the range of non-error response codes would be an easy enhancement.